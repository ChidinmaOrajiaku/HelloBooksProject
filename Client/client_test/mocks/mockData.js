import { updatePassword } from "../../actions/updatePassword";

const mockData = {
  userDetails: {
    message: 'Welcome admin',
    username: 'admin',
    token: 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJ1c2VybmFtZSI6ImRhZGR5Iiwicm9sZSI6InVzZXIiLCJpZCI6MiwiaWF0IjoxNTEzODc1ODE3LCJleHAiOjE1MTM5NjIyMTd9.oexUFhi08vhMU88LmZJvRDy9JFdThMpfb3Qz9DWWzgk'
  },
  userDetailsResponse: {
    username: 'daddy',
    role: 'user',
    id: 2,
    iat: 1513875817,
    exp: 1513962217
  },
  userCountResponse: {
    count: 7
  },
  borrowBookData: {
    bookId: 1,
    userId: 1,
  },
  borrowBookDataResponse: {
    returned: false,
    id: 24,
    userId: 1,
    bookId: 7,
    toReturnDate: '2018-01-14',
    updatedAt: '2017-12-21T21:16:08.442Z',
    createdAt: '2017-12-21T21:16:08.442Z',
    returnDate: null,
    categoryId: null
  },
  createBookData: {
    title: 'a',
    author: 'a',
    category: 'a',
    image: 'https://res.cloudinary.com/andela-chidinma/image/upload/v1511133007/g2exfx6sgbymszspybts.jpg',
    review: 'Nice'
  },
  createBookErrorData: {
    author: 'a',
    category: 'a',
    image: 'https://res.cloudinary.com/andela-chidinma/image/upload/v1511133007/g2exfx6sgbymszspybts.jpg',
    review: 'Nice'
  },
  createBookErrorResponse: 'An error occured',
  createBookResponse: {
    message: 'Succesfully added'
  },
  getABookData: {
    id: 7,
    title: 'Rage of Angels',
    author: 'Sidney Sheldon',
    category: 'Fiction',
    image: 'https://res.cloudinary.com/andela-chidinma/image/upload/v1513096198/prbdr6eghnnrcewwpodf.jpg',
    review: 'Nice Book',
    createdAt: '2017-12-12T11:11:10.823Z',
    updatedAt: '2017-12-12T16:30:00.352Z',
    userId: null,
    categoryId: null
  },
  getAllBooksData: {
    id: 7,
    title: 'Rage of Angels',
    author: 'Sidney Sheldon',
    category: 'Fiction',
    image: 'https://res.cloudinary.com/andela-chidinma/image/upload/v1513096198/prbdr6eghnnrcewwpodf.jpg',
    review: 'Nice Book',
    createdAt: '2017-12-12T11:11:10.823Z',
    updatedAt: '2017-12-12T16:30:00.352Z',
    userId: null,
    categoryId: null
  },
  getAllBorrowedBooksData: {
    id: 7,
    title: 'Rage of Angels',
    author: 'Sidney Sheldon',
    category: 'Fiction',
    image: 'https://res.cloudinary.com/andela-chidinma/image/upload/v1513096198/prbdr6eghnnrcewwpodf.jpg',
    review: 'Nice Book',
    createdAt: '2017-12-12T11:11:10.823Z',
    updatedAt: '2017-12-12T16:30:00.352Z',
    userId: null,
    categoryId: null
  },
  getCategoryData: {
    category: 'Fiction'
  },
  getUserData: {
    firstname: 'daddy',
    lastname: 'daddy',
    username: 'daddy',
    password: 'daddy',
    email: 'daddy@gmail.com'
  },
  bookData: {
    title: 'a',
    author: 'b',
    category: 'a',
    image: 'https://res.cloudinary.com/andela-chidinma/image/upload/v1511133007/g2exfx6sgbymszspybts.jpg',
    review: 'Nice'
  },
  returnBookData: {
    message: 'Successfully Returned'
  },
  userData: {
    verifyPassword: '1234',
    password: '4321'
  },
  errorMessage: 'An error occurred',
  image: 'https://res.cloudinary.com/andela-chidinma/image/upload/v1515403427/ena7dhy6xriztom6vuvo.jpg'
};

export default mockData;
