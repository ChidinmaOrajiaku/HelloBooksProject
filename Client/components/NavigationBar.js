import React from 'react';
import PropTypes from 'prop-types';
import { withRouter } from 'react-router-dom'
import { connect } from 'react-redux';
import { logout } from '../actions/signinAction'


class NavigationBar extends React.Component {

  logout(event) {
      event.preventDefault();
      this.props.logout()
      this.props.history.push("/")
  }

  componentDidMount() {
     
   }
  render() {
       
    $( document ).ready(function () {
      $(".button-collapse").sideNav();
    })

    const { logout } = this.props
    const { isAuthenticated } = this.props.auth

    const userLinks = (
        <div>
         <li><a href="/library">Library</a></li>
         <li><a href="/history">History</a></li>
         <li><a href="/profile">Profile</a></li>
         <li><a href="/admin">Admin</a></li>
         <li><a onClick={this.logout.bind(this)}>Log Out</a></li>
       </div>
    )

    const guestLinks = (
      <div>
        <li><a href="/register">Sign Up</a></li>
        <li><a href="/login">Login</a></li>
      </div>
    )
  return (
      <div className="container-fluid navbar-fixed">
         <nav className="teal">
              <div className="nav-wrapper">
                 <a href="#" className="brand-logo">HelloBooks</a>
                 <a href="#" data-activates="mobile-demo" className="button-collapse"><i className="material-icons">menu</i></a>
                 <ul className="right hide-on-med-and-down">
                 { isAuthenticated ? userLinks : guestLinks }
                 </ul>
                 <ul className="side-nav" id="mobile-demo">
                    { isAuthenticated ? userLinks : guestLinks }
                 </ul>
         </div>
       </nav>
      </div>
   );
}
};

NavigationBar.propTypes = {
  auth: PropTypes.object.isRequired,
  logout: PropTypes.func.isRequired,
}


function mapStateToProps(state) {
  return {
    auth: state.auth
  }
}

export default connect(mapStateToProps, {logout}) (withRouter(NavigationBar));
