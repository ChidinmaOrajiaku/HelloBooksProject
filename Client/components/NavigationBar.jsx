import React from 'react';
import PropTypes from 'prop-types';
import { withRouter, Link } from 'react-router-dom';
import { connect } from 'react-redux';
import image from '../img/smiley3.jpg';
import { logout } from '../actions/signinAction';
import { getUserDataRequest } from '../actions/getUser';

/**
 *
 *
 * @class NavigationBar
 * @extends {React.Component}
 */
export class NavigationBar extends React.Component {
  /**
     * @constructor
     * @param {object} props
     */
  constructor(props) {
    super(props);
    this.state = {
      loading: true,
      firstname: '',
      lastname: '',
      email: '',
    };
  }

  /**
   *
   * @returns {any} sideNav
   * @memberof NavigationBar
   */
  componentDidMount() {
    this.props.getUserDataRequest(this.props.usersId);
    $('.button-collapse').sideNav('show');
  }

  /**
   *
   * @returns {nextProps} next props
   * @param {any} nextProps
   * @memberof NavigationBar
   */
  componentWillReceiveProps(nextProps) {
    if (nextProps.getUserData) {
      this.setState({
        firstname: nextProps.getUserData.firstname,
        lastname: nextProps.getUserData.lastname,
        email: nextProps.getUserData.email
      });
    }
  }

  /**
 *
 * @returns {any} event
 * @param {any} event
 * @memberof NavigationBar
 */
  logout(event) {
    event.preventDefault();
    this.props.logout();
    this.props.history.push('/');
  }

  /**
  * React Element Markup
  * @returns {object} response object
  * @memberof NavigationBar
  */
  render() {
    const { isAuthenticated } = this.props.auth;

    const adminLinks = (
      <div>
        <li>
          <Link to="/dashboard">
            <i className="material-icons">account_circle</i>Dashboard
          </Link>
        </li>
        <li>
          <Link to="/addbooks">
            <i className="material-icons">file_upload</i>Upload Book
          </Link>
        </li>
        <li>
          <Link to="/books"><i className="material-icons">book</i>Books
          </Link>
        </li>
        <li>
          <Link to="/profile">
            <i className="material-icons">person</i>Profile
          </Link>
        </li>
        <li>
          <a onClick={this.logout.bind(this)} id="logOut">
            <i className="material-icons">fast_rewind</i>Log Out
          </a>
        </li>
      </div>
    );

    const userLinks = (
      <div>
        <li name="library">
          <Link to="/library">
            <i className="material-icons">book</i>Library
          </Link>
        </li>
        <li name="history">
          <Link to="/history"><i className="material-icons">photo_library</i>
          History
          </Link>
        </li>
        <li>
          <Link to="/profile">
            <i className="material-icons">person</i>Profile
          </Link>
        </li>
        <li name="logout">
          <a onClick={this.logout.bind(this)} id="logOut">
            <i className="material-icons">fast_rewind</i>Log Out
          </a>
        </li>
      </div>
    );
    return (
      <div className="container-fluid">
        <ul id="slide-out" className="side-nav fixed">
          <li>
            <div className="user-view">
              <a to="#" className="brand-logo">HelloBooks</a>
              <div className="background"></div>
              <a href="#!user">
                <img className="circle"
                  src={image}
                />
              </a>
              <a href="#!name">
                <span className="white-text name">
                  {`${this.state.firstname} ${this.state.lastname}`}
                </span>
              </a>
              <a href="#!email"><span className="white-text email">
                {this.state.email}</span>
              </a>
            </div>
          </li>
          {
            localStorage.username === 'admin96'
            && isAuthenticated ?
            adminLinks : userLinks }
        </ul>
        <a to="#" data-activates="slide-out" className="button-collapse">
          <i className="material-icons">menu</i>
        </a>
      </div>
    );
  }
}


NavigationBar.propTypes = {
  auth: PropTypes.object.isRequired,
  logout: PropTypes.func.isRequired,
};

/**
 *
 *
 * @param {any} state
 * @returns {state} map state to props
 */
function mapStateToProps(state) {
  return {
    auth: state.auth,
    usersId: state.auth.user.id,
    getUserData: state.getUser[0].response,
  };
}

export default connect(mapStateToProps, {
  logout,
  getUserDataRequest
})(withRouter(NavigationBar));
