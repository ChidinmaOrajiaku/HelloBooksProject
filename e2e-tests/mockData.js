module.exports = {
  file: {
    lastModified: 1508779018000,
    lastModifiedDate: 'Mon Oct 23 2017 18:16:58 GMT+0100 (WAT)',
    name: '3 card game.jpg',
    size: 642971,
    type: 'image/jpeg',
    webkitRelativePath: ''
  },
};

