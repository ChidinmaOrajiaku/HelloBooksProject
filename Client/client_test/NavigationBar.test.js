import React from 'react';
import $ from 'jquery';
import toJson from 'enzyme-to-json';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { NavigationBar } from '../components/NavigationBar.jsx';
import mockData from './mocks/mockData';

configure({ adapter: new Adapter() });

describe('<NavigationBar />', () => {
  const { userDetailsResponse } = mockData;
  const props = {
    auth: userDetailsResponse,
    logout: jest.fn(),
    getUserDataRequest: jest.fn(),
    history: {
      push: jest.fn()
    }
  };

  let mountedComponent;
  const navigationBarItem = () => {
    if (!mountedComponent) {
      mountedComponent = shallow(<NavigationBar {...props}/>);
    }
    return mountedComponent;
  };

  navigationBarItem().sideNav = jest.fn();
  $('.button-collapse').click();
  beforeEach(() => {
    mountedComponent = undefined;
  });

  it('renders <NavigationBar /> component', () => {
    expect(navigationBarItem()).toHaveLength(1);
  });

  it('should logout', () => {
    const preventDefault = jest.fn();
    navigationBarItem().find('#logOut').simulate('click', { preventDefault });
    expect(toJson(navigationBarItem())).toMatchSnapshot();
    expect(preventDefault).toBeCalled();
  });

  it('should match snapshot test', () => {
    const component = navigationBarItem();
    const tree = toJson(component);
    expect(tree).toMatchSnapshot();
  });
});
