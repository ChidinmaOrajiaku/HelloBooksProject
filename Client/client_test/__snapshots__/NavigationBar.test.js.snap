// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavigationBar /> should logout 1`] = `
<div
  className="container-fluid"
>
  <ul
    className="side-nav fixed"
    id="slide-out"
  >
    <li>
      <div
        className="user-view"
      >
        <a
          className="brand-logo"
          to="#"
        >
          HelloBooks
        </a>
        <div
          className="background"
        />
        <a
          href="#!user"
        >
          <img
            className="circle"
            src={Object {}}
          />
        </a>
        <a
          href="#!name"
        >
          <span
            className="white-text name"
          >
             
          </span>
        </a>
        <a
          href="#!email"
        >
          <span
            className="white-text email"
          />
        </a>
      </div>
    </li>
    <div>
      <li
        name="library"
      >
        <Link
          replace={false}
          to="/library"
        >
          <i
            className="material-icons"
          >
            book
          </i>
          Library
        </Link>
      </li>
      <li
        name="history"
      >
        <Link
          replace={false}
          to="/history"
        >
          <i
            className="material-icons"
          >
            photo_library
          </i>
          History
        </Link>
      </li>
      <li>
        <Link
          replace={false}
          to="/profile"
        >
          <i
            className="material-icons"
          >
            person
          </i>
          Profile
        </Link>
      </li>
      <li
        name="logout"
      >
        <a
          id="logOut"
          onClick={[Function]}
        >
          <i
            className="material-icons"
          >
            fast_rewind
          </i>
          Log Out
        </a>
      </li>
    </div>
  </ul>
  <a
    className="button-collapse"
    data-activates="slide-out"
    to="#"
  >
    <i
      className="material-icons"
    >
      menu
    </i>
  </a>
</div>
`;

exports[`<NavigationBar /> should match snapshot test 1`] = `
<div
  className="container-fluid"
>
  <ul
    className="side-nav fixed"
    id="slide-out"
  >
    <li>
      <div
        className="user-view"
      >
        <a
          className="brand-logo"
          to="#"
        >
          HelloBooks
        </a>
        <div
          className="background"
        />
        <a
          href="#!user"
        >
          <img
            className="circle"
            src={Object {}}
          />
        </a>
        <a
          href="#!name"
        >
          <span
            className="white-text name"
          >
             
          </span>
        </a>
        <a
          href="#!email"
        >
          <span
            className="white-text email"
          />
        </a>
      </div>
    </li>
    <div>
      <li
        name="library"
      >
        <Link
          replace={false}
          to="/library"
        >
          <i
            className="material-icons"
          >
            book
          </i>
          Library
        </Link>
      </li>
      <li
        name="history"
      >
        <Link
          replace={false}
          to="/history"
        >
          <i
            className="material-icons"
          >
            photo_library
          </i>
          History
        </Link>
      </li>
      <li>
        <Link
          replace={false}
          to="/profile"
        >
          <i
            className="material-icons"
          >
            person
          </i>
          Profile
        </Link>
      </li>
      <li
        name="logout"
      >
        <a
          id="logOut"
          onClick={[Function]}
        >
          <i
            className="material-icons"
          >
            fast_rewind
          </i>
          Log Out
        </a>
      </li>
    </div>
  </ul>
  <a
    className="button-collapse"
    data-activates="slide-out"
    to="#"
  >
    <i
      className="material-icons"
    >
      menu
    </i>
  </a>
</div>
`;
